### ISSUE TITLE
Move IntervalTree to factories

### ISSUE THREAD

**Issue by mccalluc** [higlass/higlass#418]
https://github.com/higlass/higlass/pull/399#discussion_r233558868

fritz writes:
> This is unrelated to the PR but it seems like IntervalTree is a factory rather than a utility

**Comment by flekschas** [comment]
**tl/dr**

My take on utils vs factories vs services is:

- factories are functions creating _objects_ with state
- utilities are stateless functions that are potentially used more than once
- services are _objects_ with state that are only initiated once. We should honestly not use them because they can create all kinds of trouble when multiple HG instances exist on the same page. React's context-provider pattern paired with a factory initiated on startup makes more sense.

Just for clarity I would move `IntervalTree` to `factories` because it's a function creating stateful objects.

Of course this is up for discussion and obviously this has low priority.
